src/main/java/Iris/Parser.java

```diff
@@ -0,0 +1,72 @@
+package iris;
+
+/**
+ * The class responsible for parsing user commands in the Iris application.
+ */
+public class Parser {
+
+    /**
+     * Parses and executes user commands.
+     *
+     * @param taskStorage The storage object for tasks.
+     * @param toDoList    The to-do list containing tasks.
+     * @param input       The user input command to be parsed.
+     */
+    public void parseCommand(Storage taskStorage, ToDoList toDoList, String input) {
```

Comment:
Perhaps can try to make multiple classes for this instead of putting it all in the Parser class. If the command later on gets too many, this file will be too big and it will be harder for you to debug and keep track of everything.